scripts/Calc_hru_wgtavg_nc.py

```diff
@@ -1,208 +1,137 @@
-          #if value of overlapping polygon is missing data -case1
-          if a == None: 
-            numvoid = numvoid+1
-            wgtArray[j] = 0 
-    
-    #Initialize Val2[ntime] 
-    Val2 = np.zeros((dim1size))
-    if np.nansum(wgtArray) > 0.0:
-    #  print (np.nansum(wgtArray) > 0.0)
-      # Adjust weight value if valid weight value (> 0) exist in list
-      newWgtArray = [x/np.nansum(wgtArray) for x in wgtArray]
-    #  print "%d invalid polygons out of %d polygons" %(numvoid,overlaps)
-    #  print "sum of original weights is %f" %np.nansum(wgtArray)
-    #  print "sum of new weights is %f" %np.nansum(newWgtArray)
-    #  print newWgtArray
-      # Go through each overlapping polygon
-      c=0
-      for j in range(len(newWgtArray)): 
-        if overlapsId[j] >= 0: # if there is at least one intersecting polygon
-
-          ij=np.where(IdVal==overlapsId[j]) # ij is tuple e.g., ij=(array[i1],array[i2]) 
-          yj = ij[0]
-          xi = ij[1]
-          Val1=newWgtArray[j]*dataVal[:,yj,xi]
-          #just check If selected hru id from runoff nc matches w/ wieght nc, if not,something wrong exit 
-          if (IdVal[yj,xi] != overlapsId[j]):
-             print "hru_id %d not exist, Sayonara" %IdVal[yj,xi]
-             sys.exit()
-          # Concatenate time series array, Val1 to oriVal containin time serise from all the intersected polygon 
-          if Val1.size: #if Val1 is not empty i.e., exclude empty value in weight list
-            c += 1
-            if c == 1:
-              oriVal = Val1
+    att_data = getattr(var, att_name)
+    return att_data
+
+
+def comp_agv_val(nc_wgt, nc_in, varname, chunk):
+    """Compute areal weighted avg value of <varname> in <nc_in>
+       for each of <chunk> hrus based on hru's weight in <nc_wgt>
+    """
+    hru_ids = get_hru_id(nc_wgt)
+    hru_data = split(hru_ids, NPROC)
+    hru_list = hru_data[chunk]
+
+    var_data = get_netCDF_data(nc_in, varname)
+    fill_val = get_netCDF_attr(nc_in, varname, '_FillValue')
+    ghruid = get_netCDF_data(nc_in, GRID_ID_NAME)
+    dim1size = var_data.shape[0]
+    # Initialize wgt_ave_val[ntime,nhru]
+    wgt_ave_val = np.full((dim1size, len(hru_list)), FILL_VALUE)
+    for i in range(len(hru_list)):
+        print hru_list[i]  # Get list of wgt for corresponding hru
+        (wgt_array, overlap_ids) = get_wgt(nc_wgt, hru_list[i])
+        # Go through wgt list and replace value with zero for following cases
+        # where overlapping polygon has missing value - case1
+        # where overlapping polygon is outside nc_wgt domain - case2
+        a = np.zeros((dim1size, len(wgt_array)))
+        for j, overlap_id in enumerate(overlap_ids):
+            # find index of grid cell that match up with hru id of overlap_ids
+            row, col = np.where(ghruid == overlap_id)
+            # if nc_in netCDF does not cover nc_wgt domain - case2
+            if not (np.size(row) and np.size(col)):
```

Comment:
are you just checking to see if these are empty arrays?

`if not row or not col` would be the cleaner way to write this.

Reply:
>are you just checking to see if these are empty arrays? 

That is correct, but did not get the same answer.

Reply:
Consider these examples:

```Python
In [13]: a = np.array([])

In [14]: b = np.array([])

In [15]: if not a or not b:
    ...:     print('neither a or b has any elements')
    ...:
neither a or b has any elements

In [16]: if not (a and b):
    ...:     print('neither a or b has any elements')
    ...:
neither a or b has any elements

In [17]: if not (a.size and b.size):
    ...:     print('neither a or b has any elements')
    ...:
neither a or b has any elements

In [18]: if not (np.size(a) and np.size(b)):
    ...:     print('neither a or b has any elements')
    ...:
neither a or b has any elements
```

If this doesn't give you what you want, I think I must be misunderstanding what you are trying to do.

Reply:
@jhamman  I did a quick test.   Difference appears if you have element zero (which is valid index for python).     I think i will use `if not (a.size and b.size):`

```python

In [2]: import numpy as np

In [3]: a = np.array([0])

In [4]: b = np.array([1])

In [5]: print(a); print(b)
[0]
[1]

In [6]: if not a or not b:
   ...:     print('neither a or b has any elements')
   ...:     
neither a or b has any elements

In [7]: if not (a.size and b.size):
   ...:     print('neither a or b has any elements')
   ...:     

In [8]: if not (a and b):
   ...:     print('neither a or b has any elements')
   ...:     
neither a or b has any elements

In [9]: if not (np.size(a) and np.size(b)):
   ...:     print('neither a or b has any elements')
   ...:     

In [10]: 

In [16]: a = np.array([0]); b = np.array([0])

In [17]: print(a); print(b)
[0]
[0]

In [18]: if not a or not b:
    ...:     print('neither a or b has any elements')
    ...:     
neither a or b has any elements

In [19]: if not (a and b):
    ...:     print('neither a or b has any elements')
    ...:     
neither a or b has any elements

In [20]: if not (a.size and b.size):
    ...:     print('neither a or b has any elements')
    ...:     

In [21]: if not (np.size(a) and np.size(b)):
    ...:     print('neither a or b has any elements')
    ...:  
```

Reply:
nice. yes, `a.size` is still an improvement.